File: apps/api/src/gpt_auto_register/core/text_import.py
from collections.abc import Iterator

CONTENT_MARKERS = {
    "=== 卡密内容 ===",
    "=== 邮箱内容 ===",
    "=== 账号内容 ===",
}
EXPORT_HEADERS = {
    "卡密导出",
    "邮箱导出",
    "账号导出",
}


def import_content_lines(text: str) -> Iterator[tuple[int, str]]:
    lines = [
        (line_number, raw_line.lstrip("\ufeff").strip())
        for line_number, raw_line in enumerate(text.splitlines(), start=1)
    ]
    marker_index = next(
        (index for index, (_, line) in enumerate(lines) if line in CONTENT_MARKERS),
        None,
    )
    candidates = lines[marker_index + 1 :] if marker_index is not None else lines
    for line_number, line in candidates:
        if not line or line in EXPORT_HEADERS or line.startswith("#"):
            continue
        yield line_number, line

File: apps/web/src/lib/text-import.ts
const CONTENT_MARKERS = new Set([
  "=== 卡密内容 ===",
  "=== 邮箱内容 ===",
  "=== 账号内容 ===",
])

const EXPORT_HEADERS = new Set(["卡密导出", "邮箱导出", "账号导出"])

export function normalizeImportText(text: string) {
  const lines = text
    .replace(/\r\n?/g, "\n")
    .split("\n")
    .map((line) => line.replace(/^\ufeff/, "").trim())
  const markerIndex = lines.findIndex((line) => CONTENT_MARKERS.has(line))
  const candidates = markerIndex >= 0 ? lines.slice(markerIndex + 1) : lines

  return candidates
    .filter(
      (line) =>
        line &&
        !EXPORT_HEADERS.has(line) &&
        !line.startsWith("#") &&
        !(line.startsWith("===") && line.endsWith("==="))
    )
    .join("\n")
}

export function importEntryCount(text: string) {
  const normalized = normalizeImportText(text)
  return normalized ? normalized.split("\n").length : 0
}
